src/lib/spinel/spinel_interface.hpp

```diff
@@ -125,21 +126,18 @@ class SpinelInterface
     /**
      * Updates the file descriptor sets with file descriptors used by the radio driver.
      *
-     * @param[in,out]  aReadFdSet   A reference to the read file descriptors.
-     * @param[in,out]  aWriteFdSet  A reference to the write file descriptors.
-     * @param[in,out]  aMaxFd       A reference to the max file descriptor.
-     * @param[in,out]  aTimeout     A reference to the timeout.
+     * @param[in,out]   aMainloopContext  A pointer to the mainloop context.
      *
      */
-    virtual void UpdateFdSet(fd_set &aReadFdSet, fd_set &aWriteFdSet, int &aMaxFd, struct timeval &aTimeout) = 0;
+    virtual void UpdateFdSet(void *aMainloopContext) = 0;
```

Comment:
We need this as `void *` so we can pass either `VirtualEvent` or `MainLoopContext` here. Is this correct?

Reply:
I am resolving this. I think that is the reason. Seems reasonable to me.